### PR TITLE
Fix agent role for push agents

### PR DESCRIPTION
**Pull request by hb9hnt**
## Pull request type

- [x] Bugfix (kindof)
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?
There is an inbound port check to the host which runs no matter whether we use pull or push agents. Also, if you reploy a push agent, initial discovery will fail, since there is no data.

Issue Number: `#437` 

## What is the new behavior?

- There is a new variable `checkmk-agent-push' which defaults to false.
- If set to `true` the port scan is skipped.
- If set to `true` we trigger a initial push using `cmk-agent-ctl push` beforce running discovery.

## Other information
As said in the issue, that's just the approach I took. I'm not saying it is a clean or general approach. It's confusing because it introduces a variable called `checkmk-agent-push` which doesn't actually activate push agents. Therefore it duplicates information that already has to be present as setting on the target folder.

### CODE REVIEW

**Comment by robin-checkmk** [comment]
recheck

File: roles/agent/defaults/main.yml [hunk 1]
```diff
@@ -31,6 +31,7 @@ checkmk_agent_folder: "{{ checkmk_folder_path | default('/') }}"
 checkmk_agent_force_foreign_changes: 'false'
 checkmk_agent_host_attributes:
   ipaddress: "{{ checkmk_agent_host_ip | default(omit) }}"
+checkmk_agent_push: false
```
Comment (robin-checkmk) [inline roles/agent/defaults/main.yml:34]: I feel like `checkmk_agent_mode: pull` would be more natural. What do you think?

**Comment by robin-checkmk** [comment]
@hb9hnt would you please sign the CLA and take a look at the failed tests? I think both are minor things, but they prevent us from moving forward. Thanks!

**Comment by hb9hnt** [comment]
I have read the CLA Document and I hereby sign the CLA or my organization already has a signed CLA.

**Comment by hb9hnt** [comment]
@robin-checkmk Sorry, I completely forgot about this pull request. I fixed the linting issue and renamed the variable according your suggestion - which I completely agree with.

File: roles/agent/tasks/Linux.yml [hunk 2]
```diff
@@ -107,3 +107,4 @@
   ansible.builtin.wait_for:
     port: 6556
     timeout: 60
+  when: checkmk_agent_mode != 'push'
```
Comment (robin-checkmk) [inline roles/agent/tasks/Linux.yml:110]: Not sure, if the conditional should be noted positively ( == pull), but as there is no objective answer, I think it is okay.

**Review by robin-checkmk — APPROVED** [review]
I will take some moments next week to think about other places, where we would need to handle the push mode. But what we got so far is fine and ready to merge. So it will definitely make the next release.

File: roles/agent/tasks/Linux.yml (hunk 2 above)
Comment (thorian93) [reply roles/agent/tasks/Linux.yml:110]: Okay, looking at this now, I realize, that there is a good reason to make this `when: checkmk_agent_mode == "pull"`, as that also covers the use-case mentioned below.

**Comment by hb9hnt** [comment]
@thorian93 / @robin-checkmk Since `checkmk_agent_mode: ssh` isn't used anywhere a hint in the readme should be enough. Maybe we could also add a comment to the `defaults/main` file.

Do you want me to add this to the PR?

**Comment by robin-checkmk** [comment]
@hb9hnt I do not want to trouble you with documentation, so I can gladly do that part. If you just change the conditional from the comment above, then we are good.

File: roles/agent/tasks/Linux.yml (hunk 2 above)
Comment (robin-checkmk) [reply roles/agent/tasks/Linux.yml:110]: @hb9hnt this one. Keeping the mode "ssh" in mind, this conditional should read when: `checkmk_agent_mode == "pull"`.

**Pull request merged**